src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_LENGTH = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[INITIAL_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == elements.length) {
+            expandMainArray();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elements.length) {
+            expandMainArray();
+        }
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound");
+        }
+        T[] tempArr1 = (T[]) new Object[index];
+        System.arraycopy(elements, 0, tempArr1, 0, index);
+        T[] tempArr2 = (T[]) new Object[size - index];
+        System.arraycopy(elements, index, tempArr2, 0, size - index);
+        elements = (T[]) new Object[++size];
+        elements[index] = value;
+        System.arraycopy(tempArr1, 0, elements, 0, index);
+        System.arraycopy(tempArr2, 0, elements, index + 1, size - index - 1);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() + size > elements.length) {
+            expandMainArray();
+        }
```

Comment:
You can remove this piece of code, as your list will grow inside add() method if needed